src/battle_util.c

```diff
@@ -3151,7 +3128,7 @@ u8 AtkCanceller_UnableToUseMove(void)
             {
                 if (Random() % 5)
                 {
-                    if (gBattleMoves[gCurrentMove].effect != EFFECT_THAW_HIT) // unfreezing via a move effect happens in case 13
+                    if (!(gBattleMoves[gCurrentMove].flags & FLAG_THAW_USER)) // unfreezing via a move effect happens in case 13
```

Comment:
I wonder if it wouldn't be better to move this check before the random chance of freezing.
As I understand it, a pokemon using a thawing move may get defrosted with two different messages, if the 20% odds are met.

Reply:
I think the second message won't be printed, as the mon has to be frozen for that code block to run and the frozen status is cleared if it thaws first in CANCELLER_FROZEN. Looks like it'd just show "Pokemon thawed out!" instead of saying the move thawed it. That's probably not how it should work though, will move the check.

Reply:
Done.